pkg/distributor/distributor.go

```diff
@@ -1013,11 +1024,13 @@ func (d *Distributor) limitsMiddleware(next push.Func) push.Func {
 
 		il := d.getInstanceLimits()
 		if il.MaxInflightPushRequests > 0 && inflight > int64(il.MaxInflightPushRequests) {
+			d.discardedInstanceRequests.WithLabelValues(validation.ReasonDistributorMaxInflightPushRequests).Inc()
 			return nil, errMaxInflightRequestsReached
 		}
 
 		if il.MaxIngestionRate > 0 {
 			if rate := d.ingestionRate.Rate(); rate >= il.MaxIngestionRate {
+				d.discardedInstanceRequests.WithLabelValues(validation.ReasonDistributorMaxIngestionRate).Inc()
```

Comment:
Should we add / extend a test to verify that the counter is incremented when these errors are hit? 

(also applies for the max-inflight push request bytes limit below and the ingester limits)

Reply:
I took a look at doing this. Some of them are racey to try to recreate in a test (ingestion rate, in-flight requests). I'd rather not test these than add something that has the potential to flake. Maybe we can revisit this in the future if we run into issues where we broke these metrics without noticing.